tape/src/main/java/com/squareup/tape/QueueElementObserver.java

```diff
@@ -0,0 +1,26 @@
+package com.squareup.tape;
+
+import java.io.IOException;
+
+public interface QueueElementObserver {
+
+  /**
+   * Observe elements flowing through the Tape queue, to make handling decisions.
+   *
+   * @param elementSize - Size of the element being added or read in bytes.
+   * @param maxElementSize - Max size of elements for this queue.
+   * @param fileContentLength - Current size of the content length as per the file header.
+   * @param isAdd - Whether the element is being added. False means element is being read.
+   * @param isInit - Whether this is during queue initialization from an existing file.
+   *
+   * @throws IOException - If observer decides to throw an exception to stop processing.
+   */
+  void observeElement(
+      int elementSize,
+      int maxElementSize,
+      int fileContentLength,
+      boolean isAdd,
+      boolean isInit) throws IOException;
```

Comment:
That's a lot of primitive params that could probably go into a separate object?

Reply:
hmm, ya may be cleaner that way, but would expand the API surface further. 6 of one, half a dozen of the other.

Reply:
after more sober thought, I think having an object will be more self-documenting. Suggestion accepted.